### PR TITLE
AD15: use current wind instead of extrapolated wind

### PR DESCRIPTION
**Pull request by andrew-platt**
This is ready for merging.

**Feature or improvement description**
When InflowWind was converted to use pointers, significant changes were required to AeroDyn to support this with linearization (InflowWind outputs are no longer inputs to AeroDyn). This is the third of 3 PR's to address this.

In this PR, the wind used in AeroDyn uses the wind interpolated from the wind data using the extrapolated position information (from AD15 inputs).  Previously the wind was extrapolated from the previous wind inputs (extrapolated like the position information).  This is a very subtle change, but affects almost all AD15 cases with turbulent winds.

old method:

- Glue code extrapolates all inputs, including position and wind
- AD15 calculations use extrapolated wind

new method:

- Glue code extrapolates position inputs
- AD15 uses extrapolated positions to calculate current wind

**Related issue, if one exists**

- PR #1882 is the first PR in the sequence.
- PR #2014 is the second PR in the sequence.

**Impacted areas of the software**

Only the AeroDyn module is affected.

**Test results, if applicable**

Several AeroDyn15 regression tests have changed as a result of using the current wind data instead of extrapolated winds.  The observed changes in the regression tests were very small for all cases except the OpenFAST MHK cases.  In general very small changes in wind velocity along the blade are observed as the sampling of the wind is now based on the extrapolated blade position (from the last 2 or 3 timesteps) instead of an extrapolation of the wind.  This leads to very small differences in wind speed at each blade or tower node, which in turn leads to very small changes blade loads (see sample plots in comments below). 

In the MHK cases, the structural response changes due to the change in loading along the blade or tower, but is very close to the original in the first few timesteps.  This is expected giving the fluid density in the MHK calculations compared to wind loading for wind turbines.

The following tests require updating:

 - OpenFAST:
    - `AWT_YFree_WSt` - sheared wind profile
    - `AOC_YFree_WTurb` - turbulent wind
    - `AOC_YFix_WSt` - sheared wind profile
    - `WP_VSP_ECD` - gust with direction change, small change in tip flap accelerations
    - `WP_VSP_WTurb` - turbulent wind, IPDefl1 slightly shifted
    - `SWRT_YFree_VS_EDG01` - gust, very tiny shift in rotor azimuth leading to shift in TipClrnc
    - `5MW_Land_DLL_WTurb`/`5MW_Land_DLL_WTurb_py` - turbulent wind
    - `5MW_OC3Mnpl_DLL_WTurb_WavesIrr`/`5MW_OC3Mnpl_DLL_WTurb_WavesIrr_Restart` - turbulent wind
    - `5MW_TLP_DLL_WTurb_WavesIrr_WavesMulti`/`5MW_TLP_DLL_WTurb_WavesIrr_WavesMulti_py` - turbulent wind
    - `5MW_OC3Spar_DLL_WTurb_WavesIrr`/`5MW_OC3Spar_DLL_WTurb_WavesIrr_py` - turbulent wind
    - `5MW_Land_BD_DLL_WTurb`/`5MW_Land_BD_DLL_WTurb_py`/`5MW_Land_BD_DLL_WTurb_cpp` - turbulent wind, small rotor azimuth drift

 - Linearization:
    - `Fake5MW_AeroLin_B1_UA4_DBEMT3` - very small changes in operating points, identical frequency/damping
    - `Fake5MW_AeroLin_B3_UA6` - very small changes in operating points, identical frequency/damping

 - AeroDyn:
    - `ad_BAR_CombinedCases` - small change in `Vindx` along the blade leading to small changes in `AxInd` and load changes
    - `ad_BAR_SineMotion` - small wind velocity changes, different loads
    - `ad_BAR_SineMotion_UA4_DBEMT3` - small `RtAeroFzh` shift from small shift in `HWindSpeedX`

 - MHK:
    - `MHK_RM1_Fixed` - small changes in tower/blade fluid induced moments at simulation start leading to changes in motion
    - `MHK_RM1_Floating` - small changes in tower/blade fluid induced moments at simulation start leading to changes in motion
    - `ad_MHK_RM1_Fixed` - minor differences in calculated `B2FldMx`
    - `ad_MHK_RM1_Floating` - shifted `HWindSpeedX` by 1 timestep ???

### CODE REVIEW

File: modules/aerodyn/src/AeroDyn_Registry.txt [hunk 1]
```diff
@@ -339,11 +340,39 @@ typedef	^	MiscVarType	ReKi            WindPos	{:}{:} - -	"XYZ coordinates to que
 typedef	^	MiscVarType	ReKi            WindVel	{:}{:} - -	"XYZ components of wind velocity"	-
 typedef	^	MiscVarType	ReKi            WindAcc	{:}{:} - -	"XYZ components of wind acceleration"	-
 
+# Inflow data storage
+typedef	^	BldInflowType	ReKi	InflowOnBlade	{:}{:}	-	-	"U,V,W at nodes on each blade (note if we change the requirement that NumNodes is the same for each blade, this will need to change)"	m/s
+typedef	^	BldInflowType	ReKi	AccelOnBlade	{:}{:}	-	-	"Wind acceleration at nodes on each blade (note if we change the requirement that NumNodes is the same for each blade, this will need to change)"	m/s
+typedef	^	RotInflowType	BldInflowType    Bld		{:}	-	-	"Blade Inputs"	-
+typedef	^	RotInflowType	ReKi	InflowOnTower	{:}{:}	-	-	"U,V,W at nodes on the tower"	m/s
```
Comment (bjonkman) [inline modules/aerodyn/src/AeroDyn_Registry.txt:347]: How about making the `BldInflowType` not just for blades? You could have this inflow type be for a single element, either blade or tower. Then you'd have RotInflowType contain an array for the blades and a scalar for the tower, something like:
 `Blade(k)%InflowVel` and `Blade(k)%InflowAccel` 
 `Tower%InflowVel` and `Tower%InflowAccel`

Comment (andrew-platt) [reply modules/aerodyn/src/AeroDyn_Registry.txt:347]: Good idea!  I'll make that change on PR #2014.

**Comment by andrew-platt** [comment]
Since this PR builds on #2014, it is not clear exactly what the difference is.  I posted a temporary PR at https://github.com/andrew-platt/openfast/pull/22 which corresponds to this PR after #2014 is merged in.

File: modules/aerodyn/src/AeroDyn_Registry.txt (hunk 1 above)
Comment (andrew-platt) [reply modules/aerodyn/src/AeroDyn_Registry.txt:347]: Actually, I'll make that on this PR (much simpler).

**Comment by andrew-platt** [comment]
**WP\_VSP\_ECD**
Screenshot 2024-05-02 at 3 41 59 PM
Changes in blade node acceleration due to small changes in sampled wind speed.

Screenshot 2024-05-02 at 3 46 11 PM
Overall agreement with prior results is good.

**ad\_BAR\_CombinedCases**
Screenshot 2024-05-02 at 3 52 24 PM
This is an artificial case with a rigid blade moving back and forth through still air while rotating.  The axial induction along the blade is slightly different.

Screenshot 2024-05-02 at 3 53 18 PM
Overall rotor power still agrees very well with prior results.

**ad\_BAR\_SineMotion**
Screenshot 2024-05-02 at 3 56 01 PM
Slight changes in loading along blade.

**Comment by andrew-platt** [comment]
The _MHK_ regression tests changed more than the wind test cases.  This is mostly due to larger loading leading to larger motions.  The start of the time series results do match well.

**MHK_RM1_Fixed**
Screenshot 2024-05-02 at 5 51 51 PM
Tower moment.

Screenshot 2024-05-02 at 6 06 29 PM
Total rotor torque moment (includes buoyancy).

**Pull request merged**